### PR TITLE
Support Live Reloading of Splits in Off-The-Grid Mode

### PR DESCRIPTION
**Pull request by HJewkes**
PR was approved last December but never merged. I have brought up to date and merged the changes to development. This PR will promote those changes to master for release as part of the 1.1.6 deployment

### CODE REVIEW

File: client/src/main/java/io/split/client/LocalhostSplitFile.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class LocalhostSplitFile extends Thread {
+    private static final Logger _log = LoggerFactory.getLogger(LocalhostSplitFile.class);
+
+    private final LocalhostSplitFactory _splitFactory;
+    private final File _file;
+    private final WatchService _watcher;
+    private AtomicBoolean _stop = new AtomicBoolean(false);
```
Comment (SenhorCastor) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: final?

Comment (SenhorCastor) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: and set it on the constructor?

File: client/src/main/java/io/split/client/LocalhostSplitFile.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+    public LocalhostSplitFile(LocalhostSplitFactory splitFactory, String directory, String fileName) throws IOException {
```
Comment (SenhorCastor) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: no checks on the input parameters?

File: client/src/main/java/io/split/client/LocalhostSplitFile.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            _log.error("Error reading file: path={}", _file.getPath(), e);
+            stopThread();
+        }
+    }
+
+    public Map<String, String> readOnSplits() throws IOException {
+        Map<String, String> onSplits = Maps.newHashMap();
+
+        BufferedReader reader = null;
+        try {
```
Comment (SenhorCastor) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: Can you us try with Autocloseable for the BufferedReader?

File: client/src/main/java/io/split/client/LocalhostSplitFile.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+                onSplits.put(feature_treatment[0], feature_treatment[1]);
+                _log.info("100% of keys will see " + feature_treatment[1] + " for " + feature_treatment[0]);
+            }
+        } catch (FileNotFoundException e) {
+            _log.warn("There was no file named " + _file.getPath() + " found. " +
+                    "We created a split client that returns default treatments for all features for all of your users. " +
+                    "If you wish to return a specific treatment for a feature, enter the name of that feature name and " +
+                    "treatment name separated by whitespace in " + _file.getPath() +
+                    "; one pair per line. Empty lines or lines starting with '#' are considered comments", e);
+        } catch (IOException e) {
```
Comment (SenhorCastor) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: why catching and throwing the exception?

File: client/src/main/java/io/split/client/LocalhostSplitFile.java (hunk 1 above)
Comment (HJewkes) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: Done

File: client/src/main/java/io/split/client/LocalhostSplitFile.java (hunk 3 above)
Comment (HJewkes) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: Done

File: client/src/main/java/io/split/client/LocalhostSplitFile.java (hunk 2 above)
Comment (HJewkes) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: Done

File: client/src/main/java/io/split/client/LocalhostSplitFile.java (hunk 4 above)
Comment (HJewkes) [inline client/src/main/java/io/split/client/LocalhostSplitFile.java]: This was a copy-paste from the code block in LocalhostSplitFactoryBuilder. Removed.

**Pull request merged**